scipy/sparse/sparsetools/dia.h

```diff
@@ -31,29 +145,86 @@ void dia_matvec(const I n_row,
                 const I n_col,
                 const I n_diags,
                 const I L,
-	            const I offsets[], 
-	            const T diags[], 
-	            const T Xx[],
-	                  T Yx[])
+                const I offsets[],
+                const T diags[],
+                const T Xx[],
+                      T Yx[])
 {
-    for(I i = 0; i < n_diags; i++){
+    for (I i = 0; i < n_diags; i++){
         const I k = offsets[i];  //diagonal offset
 
-        const I i_start = std::max<I>(0,-k);
-        const I j_start = std::max<I>(0, k);
-        const I j_end   = std::min<I>(std::min<I>(n_row + k, n_col),L);
```

Comment:
Why the removal of `std`? At least in `special`, such things caused trouble with some platform/compiler combination AFAIR.

Besides that, this PR is a massive improvement, let's try to get it into the next release.

Reply:
Well, with `using std::min, std::max;` on line 7, this can cause trouble only for very broken compilers... But in principle, the only purpose was to slightly improve readability, so if there are any objections, these `std::` prefixes can be returned here and inserted everywhere in the new code.